OrbitQt/GgpClient.h

```diff
@@ -0,0 +1,35 @@
+#ifndef ORBITQT_GGP_CLIENT_H_
+#define ORBITQT_GGP_CLIENT_H_
+
+#include <QVector>
+#include <functional>
+#include <outcome.hpp>
+#include <string>
+
+#include "GgpInstance.h"
+
+class GgpClient {
+ public:
+  // this policy means when a result is wrongly accessed, std::terminate() will
+  // be called. (.value() is accessed even though it is an error, or vice versa)
+  template <class T>
+  using ResultOrQString =
+      outcome::result<T, QString, outcome::policy::terminate>;
+
+  static ResultOrQString<GgpClient> Instantiate();
+
+  const std::string& GetVersion() const { return version_; }
+  int GetRequestsRunning() const { return requests_running_; }
+
+  using InstanceCallback =
+      std::function<void(ResultOrQString<QVector<GgpInstance>>)>;
```

Comment:
I wonder if this is better without typedef - the name hides function signature in implementation.

Reply:
It was actually only used in two places. I removed it